src/TestServer/index.js

```diff
@@ -127,8 +146,101 @@ class TestServer {
         } else {
             server = http.createServer(this._api.callback());
         }
+
         return server;
     }
+
+    _createWsServer() {
+        const wss = new ws.Server({ noServer: true });
+
+        wss.on('error', err => {
```

Comment:
will this kill the entire server on any ws error? Why not "keep calm and carry on"?

Reply:
It will. I don't actually know what the error modes are. So I don't know if the server is likely to be operable after an error. I figured that this is likely to be run in a container in a container orchestration system, and therefore that
1. it should fail loudly so that someone knows it's failing
2. the container orchestration system will handle the "keep calm and carry on" bit

Reply:
And that it will receive a PR if it needs to handle something specific.

Reply:
If it is the intention to actually kill the process at this point, setting process.exitCode will not do that by itself according to node docs. You need to do something like process.exit(1).